packages/@aws-cdk/cloud-assembly-schema/lib/assets/docker-image-asset.ts

```diff
@@ -7,7 +7,7 @@ export interface DockerImageAsset {
   /**
    * Source description for file assets
    */
-  readonly source: DockerImageSource;
+  readonly source: DockerImageSource | ExternalDockerImageSource;
```

Comment:
This is generally discouraged in jsii because it translates to Object in languages like Java. 

@RomainMuller can we somehow add a flag to jsii that will block usage of Unions with exclusions?

@iliapolo take a look at this PR

Reply:
The path of least resistance here will be to do:

```ts
readonly source?: DockerImageSource;
readonly externalSource?: ExternalDockerImageSource;
```

And have an out-of-band contract that exactly one of them must be set.